packages/firebase_performance/lib/src/performance_attributable.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of firebase_performance;
+
+/// Defines the interface that allows adding/removing attributes to any object.
+abstract class PerformanceAttributable {
```

Comment:
A little confused by the dartdoc and naming of this class. First, this is no interface, but an abstract class (or maybe I have just written too much Java). Second, the use of "_the_ interface" might seem to indicate some prior context that the reader may not have (was this interface mentioned elsewhere?). Third, the implementation is basically just a map of attributes with some specific constraints on keys and values. Fourth, there is little indication as to which "object" we are adding/removing attributes. If it is the `PerformanceAttributable` itself, then maybe this class should be defined and used as a mixin instead of an abstract base class or interface.

The member names and documentation below mentions "trace" in some places, but that seems to have little connection with the concept being defined here (until it turns out the Trace class is actually a subclass).

Maybe call this class `Attributes` or `AttributesMap`? We should name classes for what they are in and of themselves, not for how we use them in a particular situation. There seems to be nothing performance related about the API or implementation of this class.

Reply:
I named it PerformanceAttributable because it is essentially a Map wrapper that enforces the constraints from the Firebase Performance API. I added more comments to the doc to clarify that. 

I believe Performance should stay in the name of the class so that it is clear that the constraints make the class only really usable with the Performance API. I also changed it to PerformanceAttributes because Attributable sounds like an interface to me.